chain/builder.go

```diff
@@ -181,14 +181,26 @@ func BuildBlock(
 				continue
 			}
 
-			stateKeys, err := tx.StateKeys(sm)
+			readKeys, err := tx.ReadKeys(sm)
```

Comment:
I think it may be better to return a map of `StateKeys` where the value in the map is a "mode" (Read/Write/RWrite/Allocate/etc). This provides future flexibility as well to keep adding new modes that the execution engine can use.